lagranlucha-client/Assets/_Main/Scripts/Game/UI/OrderManager.cs

```diff
@@ -0,0 +1,227 @@
+﻿using System;
+using System.Linq;
+using System.Collections.Generic;
+using UnityEngine;
+
+using TMPro;
+using Zenject;
+
+namespace LaGranLucha.Game
+{
+    public class OrderManager : MonoBehaviour
+    {
+        #region FIELDS
+
+        private const float ParticlesSpawnDelay = 1.25f;
+        private const float DelayRoundStart = 1.5f;
+        private const float DelayForEndRound = 1.5f;
+
+        [Inject] private GameManager gameManager = null;
+        [Inject] private ParticlesManager particlesManager = null;
+        [Inject] private OrderUI orderUI = null;
+
+        [SerializeField] private RoundsOrders roundOrder = null;
+        [SerializeField] private List<ShownIngredient> hamburguerIngredients = null;
+        [SerializeField] private ShownIngredient friesIngredient = null;
+        [SerializeField] private ShownIngredient drinkIngredient = null;
+        [SerializeField] private TMP_Text shownName = null;
+        [SerializeField] private NumberCounter timer = null;
+        [SerializeField] private NumberCounter stars = null;
+        [SerializeField] private GameObject starParticlePrefab = null;
+        [SerializeField] private GameObject currentIngredientArrow = null;
+        [SerializeField] private OrderBar orderBar = null;
+
+        private int hamburguerStep = 0;
+        private bool completedHamburguer = false;
+        private bool completedFries = false;
+        private bool completedDrink = false;
+        private Order order = null;
+
+        #endregion
+
+        #region EVENTS
+
+        public event Action<Ingredient> onHamburguerIngredientSubmit;
+        public event Action<Ingredient> onFriesIngredientSubmit;
+        public event Action<Ingredient> onDrinkIngredientSubmit;
+        public event Action onHamburguerReset;
+        public event Action onOrderCompleted;
+        public event Action<Order> onNewOrderStart;
+
+        #endregion
+
+        #region PROPERTIES
+
+        public bool OrderComplete { get => completedHamburguer && completedFries && completedDrink; }
+        public bool HaveTimeRemaining { get => timer.IntValue > default(int); }
+        public float ThisRoundHamburguersScore { get; private set; }
```

Comment:
Just check everywhere you wrote Hamburger

Reply:
Done